enhancements/network/static-ip-addresses-vsphere.md

```diff
@@ -0,0 +1,568 @@
+
+When static IP configuration is required, Machines can not be created via MachineSets.
+The installer must create the initial set of compute Machines manually and an administrator
+must implement a `preCreate` hook for the MachineSet to allow the MachineSet to create
+Machines during day-2 operations. See [IP Configuration of Machines](#Scaling-new-Nodes-with-`machinesets`) for more information.
+
+As with the installer, the vSphere [machine reconciler](https://github.com/openshift/machine-api-operator/blob/master/pkg/controller/vsphere/reconciler.go#L745-L755) 
+will pass the static IP configuration via the `guestinfo.afterburn.initrd.network-kargs` extraconfig parameter.  
+
+
+### Day 2 Static IP Configuration
+
+Nodes being added to a cluster may be configured via an `nmstate` IP configuration or default to DHCP.  The networking configuration of a node/machine is immutable after creation. The vSphere machine API machine controller will apply the nmstate configuration when the associated VM is cloned.
+
+`machinesets` will be supported through the creation of a user-created custom controller.  This custom controller will leverage machine lifecycle hooks to
+provide IP configuration to machines descending from `machinesets` with `machine` annotated with the appropriate lifecycle hook.
+
+#### Changes Required
+
+##### Installer
+
+1. Modify the `install-config.yaml` vSphere platform specification to support nmstate configuration. The nmstate configuration is intended to be structurally compatible with the [nmstate API](https://nmstate.io/) but will omit fields that do not map logically to installation of a cluster.
+
+~~~go
+// Hosts defines `Host` configurations to be applied to nodes deployed by the installer
+type Hosts struct {
+  // Bootstrap IP configuration for the bootstrap node. If omitted, node will default to DHCP.
+  // +optional
+  Bootstrap *Host `json: "bootstrap,omitempty"`
+
+  // ControlPlane slice of IP configurations for the control plane nodes. If empty, nodes will default to DHCP.
+  // +optional
+  ControlPlane []Host `json: "controlPlane,omitempty"`
+
+  // Compute slice of IP configurations for the compute nodes. If empty, nodes will default to DHCP.
+  // +optional
+  Compute []Host `json: "compute,omitempty"`
+}
+
+// Host defines the IP configuration to be applied for a node deployed by the installer
+type Host struct {  
+  // FailureDomain refers to the name of a FailureDomain as described in https://github.com/openshift/enhancements/blob/master/enhancements/installer/vsphere-ipi-zonal.md
+  // +optional
+  FailureDomain string `json: "failureDomain"`
+
+  // NetworkConfig IP configuration to be applied
+  // +kubebuilder:validation:Required
+  NetworkConfig machineapiv1beta1.NetworkConfig string `json: "networkConfig"` 
+}
+
+// NetworkConfig Config specifies an IP configuration to be applied upon cloning.
+type NetworkConfig struct {
```

Comment:
I feel strongly that we should not do this. We're introducing yet another network API into OpenShift and there's very little benefit if we're not going to use NMState syntax. If it's a hard no on using NMState then we should skip the translation layer entirely and just allow users to pass per-node kernel args directly. That's already an established interface for network configuration so we don't introduce any more confusion than there already is and it will be much simpler to implement.

Reply:
I removed reference to nmstate/nmstate types and instead moved something closer to the [capv specification](https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/main/apis/v1beta1/types.go).  I agree that having something nmstate-like could be confusing.  But requiring raw kargs may be confusing as well.